### PR TITLE
feat: include response data in error

### LINKED ISSUE #19: include response data in onError handler
The `Form#onError` handler currently returns a string that's composed like `[${response.statusCode]: ${response.statusMessage}`, I think that's too opinionated and should be changed. The `onError` handler currently does also not include the response data. 

I see two variants:

- **onError(error: { code: string | number, message: string }, data: JSON })**

  `error`, if the error is thrown at the server-side, this will be `{ code: response.status, message: response.statusText }`. If it's a client-side error, it will be `{ code: error.name, message: error.message }`.

  `data` will be the response object from the server, send by `json(...)`

- **onError(error: { code: string | number, message: string, [key: string]: unknown })**
  The idea is to merge the code and message from the other signature, with the response data. This way there is only one error object. On error status codes, we can assume that the data that's being sent along with the response, is part of the error. Right? It also allows the user to provide the error code and message from the server.

  As a bonus, it leaves space to use `data` for the submission data. That way we could maybe also include the submitted data in the error.

### PR DESCRIPTION
**Pull request by smeijer**
Improved the `onError` callback, so that response data can be consumed:

- **onError** _(error: { code: string | number, message: string, { [key: string]: unknown }) => void_

  A callback that's invoked when the form errors on the server side. Can be used to show an error message.

  If the error originates from the server, code will default to `response.status`, and message to `response.statusText`. The error object gets merged with the json returned from the server. So the defaults can be overwritten.

  If the error originates from the client, code will be `Error.name`, and message the `Error.message`.

---

resolves #19

### CODE REVIEW

File: src/form.tsx [hunk 1]
```diff
@@ -209,24 +216,36 @@ export const Form = forwardRef(function Form(
...
               setState({
                 ...state,
                 state: 'error',
-                data: await response.json(),
-                error: `[${response.status}]: ${response.statusText}`,
+                error: {
+                  code: response.status,
+                  message: response.statusText,
+                  ...data,
+                },
```
Comment (itsMapleLeaf) [inline src/form.tsx:240]: What if we returned the whole Response object?

Comment (smeijer) [reply src/form.tsx:240]: To make the next-runtime less opinionated? Or do you see a use case?

Comment (itsMapleLeaf) [reply src/form.tsx:240]: It just seemed simpler 🤷 let the user handle the response how they want to, I suppose

Comment (smeijer) [reply src/form.tsx:240]: I'm not a big fan of having `await response.json()` in my projects. I rather have it abstracted away. 

What do you think of:

```js
error: {
  status: response.status,
  statusText: response.statusText,
  headers: response.headers,
  body: await: response.json() || response.text(),
}
```

We could do the same for onSuccess

Comment (itsMapleLeaf) [reply src/form.tsx:240]: Works for me!

**Comment by smeijer** [comment]
Closing, things will be better in #22

**Pull request closed without merging**